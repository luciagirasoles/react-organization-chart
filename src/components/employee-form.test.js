import React from "react";
import { render, fireEvent } from "@testing-library/react";

import EmployeeForm from "./employee-form";

test("EmployeeForm - no title", () => {
  const onSubmit = jest.fn();

  const { getByLabelText, getByRole, asFragment } = render(
    <EmployeeForm onSubmit={onSubmit} />
  );

  fireEvent.change(getByLabelText("Write the employee name"), {
    target: { value: "John Doe" }
  });

  fireEvent.change(getByLabelText("Write the employee title"), {
    target: { value: "Testing Engineer" }
  });

  fireEvent.submit(getByRole(/form/));

  expect(onSubmit).toHaveBeenCalledWith({
    name: "John Doe",
    title: "Testing Engineer"
  });

  expect(asFragment()).toMatchSnapshot();
});

test("EmployeeForm - title", () => {
  const onSubmit = jest.fn();

  const { getByLabelText, getByRole, asFragment } = render(
    <EmployeeForm onSubmit={onSubmit} title="CEO" />
  );

  expect(asFragment()).toMatchSnapshot();

  fireEvent.change(getByLabelText("Write the employee name"), {
    target: { value: "John Doe" }
  });

  fireEvent.submit(getByRole(/form/));

  expect(onSubmit).toHaveBeenCalledWith({
    name: "John Doe",
    title: "CEO"
  });
});
