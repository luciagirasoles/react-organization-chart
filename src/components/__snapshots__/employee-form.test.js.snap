// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeForm - no title 1`] = `
<DocumentFragment>
  .emotion-6 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  max-width: 450px;
}

.emotion-0 {
  font-size: 2rem;
  font-weight: normal;
  margin: 0;
}

.emotion-2 {
  margin: 2rem 0;
}

.emotion-1 {
  background: none;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  box-sizing: border-box;
  display: block;
  font-size: 1rem;
  padding: .5rem;
  width: 100%;
}

.emotion-1:focus {
  outline: none;
  border-color: #444444;
}

.emotion-5 {
  background-color: black;
  border: 1px solid black;
  border-radius: .25rem;
  color: white;
  cursor: pointer;
  font-size: .8rem;
  padding: .75rem 0;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.emotion-5:hover {
  background-color: white;
  color: black;
}

<div
    class="emotion-6"
  >
    <form>
      <h1
        class="emotion-0"
      >
        's Employee Name
      </h1>
      <div
        class="emotion-2"
      >
        <input
          aria-label="Write the employee name"
          class="emotion-1"
          name="name"
          placeholder="John Doe"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="emotion-2"
      >
        <input
          aria-label="Write the employee title"
          class="emotion-1"
          name="title"
          placeholder="Software Engineer"
          type="text"
          value=""
        />
      </div>
      <button
        class="emotion-5"
      >
        Create employee
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`EmployeeForm - title 1`] = `
<DocumentFragment>
  .emotion-4 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  max-width: 450px;
}

.emotion-0 {
  font-size: 2rem;
  font-weight: normal;
  margin: 0;
}

.emotion-2 {
  margin: 2rem 0;
}

.emotion-1 {
  background: none;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  box-sizing: border-box;
  display: block;
  font-size: 1rem;
  padding: .5rem;
  width: 100%;
}

.emotion-1:focus {
  outline: none;
  border-color: #444444;
}

.emotion-3 {
  background-color: black;
  border: 1px solid black;
  border-radius: .25rem;
  color: white;
  cursor: pointer;
  font-size: .8rem;
  padding: .75rem 0;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.emotion-3:hover {
  background-color: white;
  color: black;
}

<div
    class="emotion-4"
  >
    <form>
      <h1
        class="emotion-0"
      >
        's CEO Name
      </h1>
      <div
        class="emotion-2"
      >
        <input
          aria-label="Write the employee name"
          class="emotion-1"
          name="name"
          placeholder="John Doe"
          required=""
          type="text"
          value=""
        />
      </div>
      <button
        class="emotion-3"
      >
        Create employee
      </button>
    </form>
  </div>
</DocumentFragment>
`;
